OpenRA.Mods.Common/Traits/Buildings/Exit.cs

```diff
@@ -23,6 +26,9 @@ public class ExitInfo : TraitInfo<Exit>, Requires<IOccupySpaceInfo>
 		public readonly CVec ExitCell = CVec.Zero;
 		public readonly int Facing = -1;
 
+		[Desc("Type tags on this exit.")]
+		public readonly HashSet<string> Types = new HashSet<string>();
```

Comment:
It means changing all the rules again, but `Types` &rarr; `ProductionTypes` would be good here too to make it clear.